src/api/libcellml/generatorprofile.h

```diff
@@ -72,6 +72,15 @@ class LIBCELLML_EXPORT GeneratorProfile
      */
     Profile profile() const;
 
+    /**
+     * @brief Get the @c Profile as a string for this @c GeneratorProfile.
+     *
+     * Return the @c Profile as a string for this @c GeneratorProfile.
+     *
+     * @return The @c Profile as a string for this @c GeneratorProfile.
+     */
+    std::string profileAsString() const;
```

Comment:
Is this more getting the profile name as a string? I'd half expect that getting the profile as a string would return the profile definition...

Reply:
What do you mean by the profile definition? A profile contains a lot, I mean a lot, of information...

Reply:
exactly, so if asking for a profile as a string I'd expect to get a string version of all that information. Whereas the `C` or `Python` is just the name of the profile..

Reply:
I agree that the resolution of this issue is about converting enums to strings, not about poorly named enums.  That is a different can of worms.